user.py

```diff
@@ -0,0 +1,360 @@
+
+    def __is_la_str(self):
+        return self.get_details().get("custom:is_la", "0")
+
+    def is_la(self):
+        return True if self.__is_la_str() == "1" else False
+
+    """
+    Create and set properties of cognito user
+
+    Create user in cognito user pool
+    If successful then:
+    - Set MFA preference to SMS
+    - Set MFA SMS phone number
+    - Add user to the requested cognito group
+    """
+
+    def create(self, name, phone_number, custom_paths, is_la, group_name):
+        phone_number = self.sanitise_phone(phone_number)
+        if not self.email_address_is_valid():
+            return False
+        if phone_number == "":
+            return False
+        if not self.user_paths_are_valid(is_la, custom_paths, group_name):
+            return False
+        user_created = create_user(
+            name, self.email_address, phone_number, is_la, custom_paths
+        )
+
+        if user_created:
+            return True
+
+        set_mfa = self.set_mfa_preferences()
+        set_settings = self.set_user_settings()
+        added_to_group = self.add_to_group(group_name)
+        return set_mfa and set_settings and added_to_group
+
+    def set_mfa_preferences(self):
+        additional_arguments = {
+            "SMSMfaSettings": {"Enabled": True, "PreferredMfa": True},
+        }
+        return make_request(
+            "admin_set_user_mfa_preference", self.email_address, additional_arguments
+        )
+
+    def set_user_settings(self):
+        additional_arguments = {
+            "MFAOptions": [{"DeliveryMedium": "SMS", "AttributeName": "phone_number"}],
+        }
+        return make_request(
+            "admin_set_user_settings", self.email_address, additional_arguments
+        )
+
+    def add_to_group(self, group_name=None):
+        if group_name is None:
+            group_name = "standard-download"
+        if group_name not in get_group_map().keys():
+            return False
+
+        return make_request(
+            "admin_add_user_to_group", self.email_address, {"GroupName": group_name}
+        )
+
+    def set_group(self, new_group_name):
+        if new_group_name is None:
+            return
+        if not isinstance(new_group_name, str):
+            raise ValueError("ERR: %s: new_group_name is not str")
+
+        group_name = self.details["group"]["value"]
+        if group_name != new_group_name:
+            make_request(
+                "admin_remove_user_from_group",
+                self.email_address,
+                {"GroupName": group_name},
+            )
+            self.add_to_group(new_group_name)
+
+    def sanitise_phone(self, phone_number):
+        if phone_number != "":
+            phone_number = re.sub(r"[^0-9]", "", phone_number)
+            if phone_number.startswith("0"):
+                phone_number = "+44" + phone_number[1:]
+            if phone_number.startswith("44"):
+                phone_number = "+44" + phone_number[2:]
+            return phone_number
```

Comment:
Do we want to validate if the phone number starts with 7 or +44? What would happen in those scenarios?

I guess I'm trying to anticipate copy/paste errors more than anything else, especially in a scenario where this is rolled out to more than just a few admin users.

Reply:
I think I see what you mean. I think it would be a "nice thing" to do but I'm not too worried about doing it immediately as there is so much scope to get phone numbers wrong I think once we start tugging on that thread we'll get into a bit of a rabbit hole - perhaps if there is a lot of feedback about phone validations it might be something to add but it might not really be much of a problem

Reply:
I think you're write. If it becomes a problem we can fix it.